azure-pipelines.yml

```diff
@@ -87,21 +87,37 @@ steps:
   displayName: 'lint samples'
   condition: and(succeeded(), ne(variables.shouldPublish, true))
 
-  #
-  # LERNA VERSION
   # * Configure git with a commit user
   # * Update git remote to include a Personal Access Token, so lerna can push
   # * Explicitly check out the current branch since Pipelines will use disconnected HEAD by default
-  # * Increment pre-release version
-  #     * Force version update even if lerna doesn't detect changes
-  #     * Include [skip ci] in commit message to stop pipeline from triggering again when lerna pushes
-  # * Only run for 'dev' branch
-  #
+  # * Only run for 'dev' and 'release' branches
 - script: |
     git config user.email builds@sitecore.com
     git config user.name "Automated Build"
     git remote set-url origin https://$GITHUB_PAT_ENV@github.com/Sitecore/jss.git
     git checkout $(branchName)
+  displayName: 'apply git configuration'
+  condition: and(succeeded(), eq(variables.shouldPublish, true))
+  env:
+    GITHUB_PAT_ENV: $(GITHUB-PAT)
+
+  # Only run for 'dev' and 'release' branches
+  # in order to don't produce extra changes in the Pull Request
+- script: |
+    yarn generate-docs
+    git add ./ref-docs/**/*
+    git commit -m "Update API docs [skip ci]"
+  displayName: 'generate API docs'
+  condition: and(succeeded(), eq(variables.shouldPublish, true))
+
+  #
+  # LERNA VERSION
+  # * Increment pre-release version
+  #     * Force version update even if lerna doesn't detect changes
+  #     * Include [skip ci] in commit message to stop pipeline from triggering again when lerna pushes
+  # * Only run for 'dev' and 'release' branches
+  #
+- script: |
     $(npm bin)/lerna version prerelease --preid canary --force-publish --message "version %s [skip ci]" --yes
   displayName: 'pre-release version update'
   condition: and(succeeded(), eq(variables.shouldPublish, true))
```

Comment:
Can get rid of the GITHUB_PAT_ENV from this step's envs (below)

Reply:
Yeah, it's not needed here